swarm/src/behaviour.rs

```diff
@@ -449,10 +447,12 @@ pub enum FromSwarm<'a, Handler> {
     ListenerError(ListenerError<'a>),
     /// Informs the behaviour that a listener closed.
     ListenerClosed(ListenerClosed<'a>),
-    /// Informs the behaviour that we have discovered a new external address for us.
-    NewExternalAddr(NewExternalAddr<'a>),
-    /// Informs the behaviour that an external address was removed.
-    ExpiredExternalAddr(ExpiredExternalAddr<'a>),
+    /// Informs the behaviour that we have discovered a new candidate for an external address for us.
+    NewExternalAddrCandidate(NewExternalAddrCandidate<'a>),
```

Comment:
This and the events below miss their corresponding copies in `SwarmEvent`, correct @thomaseizinger? Or was it a deliberate choice to not report these events to the user as well?

Related to https://github.com/libp2p/rust-libp2p/discussions/4688.

Reply:
It wasn't a deliberate choice to omit them but I have to admit I also didn't know that we were activel trying to keep these in sync.